akka-javasdk/src/main/scala/akka/javasdk/impl/SdkRunner.scala

```diff
@@ -331,6 +331,10 @@ private object ComponentLocator {
       val serviceSetupClassName = descriptorConfig.getString(DescriptorServiceSetupEntryPath)
       val serviceSetup = system.dynamicAccess.getClassFor[AnyRef](serviceSetupClassName).get
       if (serviceSetup.hasAnnotation[Setup]) {
+        if (!classOf[ServiceSetup].isAssignableFrom(serviceSetup)) {
+          throw new IllegalStateException(
+            "A class [" + serviceSetup + "] annotated with @Setup must implement [akka.javasdk.ServiceSetup] interface")
```

Comment:
We can do this in the annotation processor, no? 

Then this error shows up at compile time.

Reply:
If we don't have it already, we can take the opportunity to throw an error if there is more than one class annotated with Setup

Reply:
Ok, let's do it in another PR. 
But it must already depend on the SDK. It has access to the annotations already.